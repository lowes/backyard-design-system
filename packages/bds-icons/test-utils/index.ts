import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
