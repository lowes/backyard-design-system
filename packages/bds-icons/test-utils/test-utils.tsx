import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
