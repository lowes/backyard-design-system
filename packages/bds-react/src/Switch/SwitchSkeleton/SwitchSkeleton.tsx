import * as React from 'react'
import classnames from 'classnames'
import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'

import SwitchSkeletonWrapper from './styles'

import Skeleton from '../../Skeleton'

/**
 * Backyard's Switch Skeleton component
 * 
 * Returns a pill shaped Skeleton rect of a predefined height and width.
 * These can not be overwritten. Optional boolean parameter of `withLabel`
 * to display a Skeleton rect as text next to the switch.
 */

const SwitchSkeleton: BDSForwardRef<SwitchSkeletonProps> = React.forwardRef<
    SwitchSkeletonRef,
    SwitchSkeletonProps
>(function SwitchSkeleton (
    {
        size = 'large',
        withLabel = false,
        animate = false,
        className,
        ...props
    },
    ref,
) {



    return (
        <SwitchSkeletonWrapper className={classnames(className, 'switch--skeleton')} {...props} ref={ref}>
            <Skeleton 
                variant='rect'
                height={(size === 'small') ? 'size_16' : 'size_24'}
                width={(size === 'small') ? '2.125rem' : '2.75rem'}
                animate={animate}
                style={{borderRadius: '500px'}}
            />
            {
                (withLabel) ?
                    <Skeleton 
                        variant='rect'
                        height='size_20'
                        width='size_84'
                        animate={animate}
                        className='skeleton--label'
                    /> :
                    null
            }
        </SwitchSkeletonWrapper>
    )
})

type SwitchSkeletonRef = HTMLDivElement

interface SwitchSkeletonProps {
    animate?: boolean 
    size?: string
    withLabel?: boolean
    className?: string
}

SwitchSkeleton.bdsName = 'SwitchSkeleton'

export { SwitchSkeleton }

export type { SwitchSkeletonProps, SwitchSkeletonRef }

export default SwitchSkeleton