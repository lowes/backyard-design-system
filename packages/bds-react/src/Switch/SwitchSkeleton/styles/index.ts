import styled from 'styled-components'

const SwitchSkeletonWrapper = styled.div`
    display: inline-flex;
    flex-direction: row;
    align-items: center;

    .skeleton--label {
        margin-left: ${({ theme }) => theme.sizes.size_16};
    }
`

export { SwitchSkeletonWrapper }
export default SwitchSkeletonWrapper