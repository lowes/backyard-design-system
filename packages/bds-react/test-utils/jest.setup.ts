import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
