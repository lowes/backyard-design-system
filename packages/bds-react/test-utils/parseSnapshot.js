const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
