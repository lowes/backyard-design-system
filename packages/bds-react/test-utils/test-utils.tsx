import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
