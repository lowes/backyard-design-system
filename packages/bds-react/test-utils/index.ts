// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
