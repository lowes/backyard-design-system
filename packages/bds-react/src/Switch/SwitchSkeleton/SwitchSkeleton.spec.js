import * as React from 'react'
import { render } from '../../../test-utils'
import SwitchSkeleton from './SwitchSkeleton'

describe('SwitchSkeleton Snapshots', () => {
    test('SwitchSkeleton default snapshot', () => {
        const { asFragment } = render(
            <SwitchSkeleton />
        )

        expect(asFragment()).toMatchSnapshot()
    })


    test('SwitchSkeleton animate snapshot', () => {
        const { asFragment } = render(
            <SwitchSkeleton animate={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('SwitchSkeleton with label', () => {
        const { asFragment } = render(
            <SwitchSkeleton withLabel={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})