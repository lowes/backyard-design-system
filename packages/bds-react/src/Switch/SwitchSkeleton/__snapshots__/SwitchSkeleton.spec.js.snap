// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSkeleton Snapshots SwitchSkeleton animate snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .skeleton--label {
  margin-left: 1rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 2.75rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 2.75rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 switch--skeleton"
  >
    <div
      class="c1 variant--rect shape--rounded style--animated"
      height="size_24"
      style="border-radius: 500px;"
      width="2.75rem"
    />
  </div>
</DocumentFragment>
`;

exports[`SwitchSkeleton Snapshots SwitchSkeleton default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .skeleton--label {
  margin-left: 1rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 2.75rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 2.75rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 switch--skeleton"
  >
    <div
      class="c1 variant--rect shape--rounded style--default"
      height="size_24"
      style="border-radius: 500px;"
      width="2.75rem"
    />
  </div>
</DocumentFragment>
`;

exports[`SwitchSkeleton Snapshots SwitchSkeleton with label 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .skeleton--label {
  margin-left: 1rem;
}

.c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 2.75rem;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 2.75rem;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c2 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c2.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c2.variant--circle {
  width: 5.25rem;
  height: 1.25rem;
  border-radius: var(--bds-border-radius-circle);
}

.c2.variant--rect {
  width: 5.25rem;
  height: 1.25rem;
}

.c2.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--rect.shape--squared {
  border-radius: 0;
}

.c2.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c2.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c2.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 switch--skeleton"
  >
    <div
      class="c1 variant--rect shape--rounded style--default"
      height="size_24"
      style="border-radius: 500px;"
      width="2.75rem"
    />
    <div
      class="c2 variant--rect shape--rounded skeleton--label style--default"
      height="size_20"
      width="size_84"
    />
  </div>
</DocumentFragment>
`;
